spec/support/specialist_publisher_assertion_helpers.rb

```diff
@@ -31,9 +31,17 @@ def expect_error(message)
     end
   end
 
+  def expect_preview_draft_link
+    expect(page).to have_link("Preview draft")
+  end
+
   def expect_unpublished
     expect(find(".alert").text).to match(/^Unpublished/)
   end
 
+  def expect_discarded_draft
+    expect(find(".alert").text).to match(/^Discarded/)
```

Comment:
Would be good to have a more specific expectation here, eg verifying that the row for the document itself now has a state of "published" rather than "published with new draft", or in the case of a never-published item, that it doesn't appear in the list at all.